packages/web-components/src/components/dotcom-shell/dotcom-shell-container.ts

```diff
@@ -0,0 +1,256 @@
+import { html, property, customElement, LitElement } from 'lit-element';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings';
+import { BasicLink, BasicLinkSet, MastheadLink } from '../../globals/services-store/types/translateAPI';
+import { USER_AUTHENTICATION_STATUS } from '../../globals/services-store/types/profileAPI';
+import { FOOTER_SIZE } from '../footer/footer';
+import '../footer/footer-container';
+import { MastheadProfileItem } from '../masthead/masthead-container';
+import { LocaleModalLocaleList } from '../locale-modal/locale-modal-composite';
+import './dotcom-shell';
+import styles from './dotcom-shell-container.scss';
+
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * Dotcom shell.
+ *
+ * @element dds-dotcom-shell-container
+ */
+@customElement(`${ddsPrefix}-dotcom-shell-container`)
+class DDSLinkWithIcon extends LitElement {
+  private _footerRenderRoot: Element | null = null;
+
+  private _mastheadRenderRoot: Element | null = null;
+
+  private _createFooterRenderRoot() {
+    const footer = this.ownerDocument.createElement(`${ddsPrefix}-footer-container`);
+    this.parentNode?.insertBefore(footer, this.nextSibling);
+    return footer;
+  }
+
+  private _createMastheadRenderRoot() {
+    const masthead = this.ownerDocument.createElement(`${ddsPrefix}-masthead-container`);
+    this.parentNode?.insertBefore(masthead, this);
+    return masthead;
+  }
+
+  /**
+   * `true` to activate the search box.
+   */
+  @property({ attribute: 'activate-search' })
+  activateSearch = false;
+
+  /**
+   * The profile items for authenticated state.
+   */
+  @property({ attribute: false })
+  authenticateProfileItems?: MastheadProfileItem[];
+
+  /**
+   * The brand name.
+   */
+  @property({ attribute: 'brand-name' })
+  brandName?: string;
+
+  /**
+   * The g11n collator to use for sorting contry names.
+   */
+  @property({ attribute: false })
+  collatorCountryName = new Intl.Collator();
+
+  /**
+   * The throttle timeout to run query upon user input.
+   */
+  @property({ type: Number })
+  inputTimeout?: number;
+
+  /**
+   * The footer links.
+   */
+  @property({ attribute: false })
+  footerLinks: BasicLinkSet[] = [];
+
+  /**
+   * The language to show in the UI.
+   */
+  @property({ attribute: 'lang-display' })
+  langDisplay?: string;
+
+  /**
+   * The language used for query.
+   */
+  @property()
+  language?: string;
+
+  /**
+   * The legal nav links.
+   */
+  @property({ attribute: false })
+  legalLinks: BasicLink[] = [];
+
+  /**
+   * The locale list.
+   */
+  @property({ attribute: false })
+  localeList?: LocaleModalLocaleList;
+
+  /**
+   * The nonce used for logging in.
+   */
+  @property({ attribute: 'login-nonce' })
+  loginNonce?: string;
+
+  /**
+   * The link target to the masthead logo.
+   */
+  @property({ attribute: 'logo-href' })
+  logoHref?: string;
+
+  /**
+   * The `aria-label` attribute for the top-level container.
```

Comment:
@asudoh I wonder if we should have the word `aria` in the attribute name so it's clear that is what it is for, like `ariaMastheadContainer`, `ariaMenuBar`, etc...

Reply:
Interesting point, the wording came from Carbon core library, if we want to change the wording pattern we can go over the codebase and change that.

Reply:
I think adding `aria` somewhere in the property name makes sense to me, as that is its main intent.

Reply:
Another thought here; Neowsers have native `aria-` attribute and if we define our own attributes beginning with `aria-` I imagine our users will confuse them with native ones.

Reply:
ok, I think that label sounds fine!

Reply:
Cool - Updated.